src/app/reporting/EventLoggingTypes.h

```diff
@@ -0,0 +1,187 @@
+    First = 1,
+    /**
+     * Critical priority denotes events whose loss would
+     * directly impact customer-facing features. Applications may use
+     * loss of Production Critical events to indicate system failure.
+     * On constrained devices, entries logged with Critical
+     * priority must be accounted for in the power and memory budget,
+     * as it is expected that they are always logged and offloaded
+     * from the device.
+     */
+    Critical = First,
+    /**
+     * Info priority denotes log entries that provide extra insight
+     * and diagnostics into the running system. Info logging level may
+     * be used over an extended period of time in a production system,
+     * or may be used as the default log level in a field trial. On
+     * the constrained devices, the entries logged with Info level must
+     * be accounted for in the bandwidth and memory budget, but not in
+     * the power budget.
+     */
+    Info = 2,
+
+    /**
+     *  Debug priority denotes log entries of interest to the
+     *  developers of the system and is used primarily in the
+     *  development phase. Debug priority logs are
+     *  not accounted for in the bandwidth or power budgets of the
+     *  constrained devices; as a result, they must be used only over
+     *  a limited time span in production systems.
+     */
+    Debug   = 3,
+    Last    = Debug,
+    Invalid = Last + 1,
+
+};
+
+/**
+ * @brief
+ *   The structure that provides a full resolution of the cluster.
+ */
+struct EventSchema
+{
+    EventSchema(chip::NodeId aNodeId, chip::EndpointId aEndpointId, chip::ClusterId aClusterId, chip::EventId aEventId,
+                PriorityLevel aPriority) :
+        mNodeId(aNodeId),
+        mEndpointId(aEndpointId), mClusterId(aClusterId), mEventId(aEventId), mPriority(aPriority){};
+    chip::NodeId mNodeId         = 0;
+    chip::EndpointId mEndpointId = 0;
+    chip::ClusterId mClusterId   = 0;
+    chip::EventId mEventId       = 0;
+    PriorityLevel mPriority      = PriorityLevel::Invalid;
+};
+
+/**
+ * @brief
+ *   The union that provides an application set system or UTC timestamp.
+ */
+struct Timestamp
+{
+    enum class Type
+    {
+        kInvalid = 0,
+        kSystem,
+        kUTC
+    };
+    Timestamp(Type aType) : mType(aType) { mValue = 0; };
+    Timestamp(Type aType, uint64_t aValue) : mType(aType), mValue(aValue){};
+    void Init(Type aType, uint64_t aValue)
+    {
+        mType  = aType;
+        mValue = aValue;
+    }
+    static Timestamp UTC(uint64_t aValue)
+    {
+        Timestamp timestamp(Type::kUTC, aValue);
+        return timestamp;
+    }
+    static Timestamp System(uint64_t aValue)
+    {
+        Timestamp timestamp(Type::kSystem, aValue);
+        return timestamp;
+    }
+
+    Type mType      = Type::kInvalid;
+    uint64_t mValue = 0;
+};
+
+/**
+ *   The structure that provides options for the different event fields.
+ */
+class EventOptions
+{
+public:
+    EventOptions(void);
```

Comment:
Please add initializers in the header.

Looking at the class it is unclear to me what EventOptions(void) will do (what will the timestamp be? will it be urgent?) Same for EventOptions(timestamp) - is it urgent by default or not?

Consider having static methods. Much easier to read:

```cpp
CallFoo(EventOptions::Urgent(timestamp))
```

than

```cpp
CallFoo(EventOptions(timestamp, true));
```